mu-plugins/blocks/site-breadcrumbs/src/view.js

```diff
@@ -0,0 +1,191 @@
+ * @param {HTMLElement}        container  Breadcrumb parent container
+ * @param {number}             breakpoint The breakpoint in pixels.
+ *
+ * @return {void}
+ */
+const collapseCrumbs = ( arr, container, breakpoint ) => {
+	const middleCrumbs = arr.slice( 1, arr.length - 1 );
+
+	// First, try to truncate the text
+	let index = 0;
+	while ( getAvailableSpace( container ) < breakpoint && ! allTruncated( middleCrumbs ) ) {
+		middleCrumbs[ index ].firstChild.classList.add( 'is-truncated' );
+		middleCrumbs[ index ].firstChild.firstChild.classList.add( 'screen-reader-text' );
+		index++;
+	}
+
+	// Second, hide the items if everything is truncated
+	let index2 = 0;
+	while ( getAvailableSpace( container ) < breakpoint && ! allHidden( middleCrumbs ) ) {
+		middleCrumbs[ index2 ].classList.add( 'is-hidden' );
+		index2++;
+	}
+
+	const remainingSpace = getAvailableSpace( container );
+	// if we truncated and hid all the items, truncate the last part
+	if ( remainingSpace < breakpoint ) {
+		const lastPart = arr[ arr.length - 1 ];
+		const width = Math.max( 75, lastPart.getBoundingClientRect().width - breakpoint );
+		lastPart.style.width = `${ width }px`;
+	}
+};
+
+/**
+ * This function expands the breadcrumbs until the available space is less than the breakpoint.
+ *
+ * The expected html for this to work is:
+ * <span>
+ *     <a href="#" data-title="Breadcrumb Title">
+ *         <span>Text</span>
+ *     </a>
+ * </span>
+ *
+ * @param {Array<HTMLElement>} arr                     List of breadcrumb elements.
+ * @param {HTMLElement}        container               Breadcrumb parent container
+ * @param {number}             breakpoint              The breakpoint in pixels.
+ * @param {number}             finalPartOriginalLength The original width of the last crumb.
+ *
+ * @return {void}
+ */
+const expandCrumbs = ( arr, container, breakpoint, finalPartOriginalLength ) => {
+	const middleCrumbs = arr.slice( 1, arr.length - 1 );
+	const currentSpaceValue = getAvailableSpace( container );
+	let pixelToAllocate = Math.ceil( currentSpaceValue - breakpoint );
+
+	// If the last part has ellipses, expand it.
+	const lastPart = arr[ arr.length - 1 ];
+	const currWidth = parseInt( lastPart.style.width );
+
+	if ( currWidth < finalPartOriginalLength ) {
+		const newWidth = Math.min( currWidth + pixelToAllocate, finalPartOriginalLength );
+		lastPart.style.width = `${ newWidth }px`;
+		pixelToAllocate -= newWidth - currWidth;
+	}
+
+	// 20 is roughly the width of the ellipsis.
+	if ( pixelToAllocate < 20 ) {
+		return;
+	}
+
+	/**
+	 * If there are hidden elements, show them first.
+	 */
+	const hiddenEls = middleCrumbs.filter( ( crumb ) => crumb.classList.contains( 'is-hidden' ) );
+
+	if ( hiddenEls.length ) {
+		hiddenEls[ 0 ].classList.remove( 'is-hidden' );
+		return;
+	}
+
+	/**
+	 * Loop through right to left, expand
+	 */
+	for ( let i = middleCrumbs.length - 1; i >= 0; i-- ) {
+		const anchorElement = middleCrumbs[ i ].firstChild;
```

Comment:
IMO this is easier to read

```suggestion
	const crumbsRightToLeft = middleCrumbs.slice().reverse();
	crumbsRightToLeft.forEach( ( { firstChild: anchorElement  } ) => {
```

Reply:
I feel like the fact that you needed to explain how `slice` and `reverse` works, maybe suggests it isn't otherwise easier to understand? :) 

I don't feel strongly either way.

Reply:
Yeah, looking at this again, creating a copy probably isn't the most performant suggestion.